architecture.md

```diff
@@ -1,11 +1,73 @@
-# MongoDB Community Operator Architecture
+# MongoDB Community Kubernetes Operator Architecture
 
-**Work in Progress**
+The MongoDB Community Kubernetes Operator is a [Custom Resource Definition](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/) and a [Controller](https://kubernetes.io/docs/concepts/architecture/controller/).
 
-## Topics
+## Table of Contents
 
-* Automation Agent
-* Cluster Configuration and ConfigMap
-* MongoDB Docker Images
-* Example case: MongoDB version upgrade
+- [Cluster Configuration](#cluster-configuration)
+- [Example: MongoDB Version Upgrade](#example-mongodb-version-upgrade)
+- [MongoDB Docker Images](#mongodb-docker-images)
 
+## Cluster Configuration
+
+You create and update MongoDB resources by defining a MongoDB resource definition. When you apply the MongoDB resource definition to your Kubernetes environment, the Operator:
+
+1. Creates a [StatefulSet](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/) that contains one [pod](https://kubernetes.io/docs/concepts/workloads/pods/pod-overview/) for each [replica set](https://docs.mongodb.com/manual/replication/) member. 
+2. Creates two [containers](https://kubernetes.io/docs/concepts/containers/overview/) in each pod: 
+   - A container for the [`mongod`](https://docs.mongodb.com/manual/reference/program/mongod/index.html) process binary.
+   - A container for the Automation Agent. 
```

Comment:
I'm missing a bit of information about what the Automation agent it.

Reply:
Good idea, reworded slightly & moved up.